platform/posix/transport/src/openssl_posix.c

```diff
@@ -0,0 +1,674 @@
+ *
+ * OpenSSL does not provide a single function for reading and loading certificates
+ * from files into stores, so the file API must be called. Start with the
+ * root certificate.
+ *
+ * @param[out] pSslSetup Destination for the trusted server root CA.
+ * @param[in] pRootCaPath Filepath string to the trusted server root CA.
+ * @param[in] rootCaPathLen brief Length associated with trusted server root CA.
+ *
+ * @return 1 on success; otherwise, failure;
+ */
+static int setRootCa( SSL_CTX * pSslSetup,
+                      const char * pRootCaPath,
+                      size_t rootCaPathLen );
+
+/**
+ * @brief Set X509 certificate as client certificate for the server to authenticate.
+ *
+ * @param[out] pSslSetup Destination for the client certificate.
+ * @param[in] pClientCertPath Filepath string to the client certificate.
+ * @param[in] clientCertPathLen brief Length associated with client certificate.
+ *
+ * @return 1 on success; otherwise, failure;
+ */
+static int setClientCertificate( SSL_CTX * pSslSetup,
+                                 const char * pClientCertPath,
+                                 size_t clientCertPathLen );
+
+/**
+ * @brief Set private key for the client's certificate.
+ *
+ * @param[out] pSslSetup Destination for the private key.
+ * @param[in] pPrivateKeyPath Filepath string to the client certificate's private key.
+ * @param[in] privateKeyPathLen brief Length associated with client certificate's private key.
+ *
+ * @return 1 on success; otherwise, failure;
+ */
+static int setPrivateKey( SSL_CTX * pSslSetup,
+                          const char * pPrivateKeyPath,
+                          size_t privateKeyPathLen );
+
+/**
+ * @brief Passes TLS credentials to the OpenSSL library.
+ *
+ * Provides the root CA certificate, client certificate, and private key to the
+ * OpenSSL library. If the client certificate or private key is not NULL, mutual
+ * authentication is used when performing the TLS handshake.
+ *
+ * @param[out] pSslSetup Destination for the imported credentials.
+ * @param[in] pOpensslCredentials TLS credentials to be validated.
+ *
+ * @return 1 on success; otherwise, failure;
+ */
+static int setCredentials( SSL_CTX * pSslSetup,
+                           const OpensslCredentials_t * pOpensslCredentials );
+
+/**
+ * @brief Set optional configurations for the TLS connection.
+ *
+ * This function is used to set SNI, MFLN, and ALPN protocols.
+ *
+ * @param[in] pSslContext Destination for the imported credentials.
+ * @param[in] pOpensslCredentials TLS credentials containing configurations.
+ */
+static void setOptionalConfigurations( SSL * pSslContext,
+                                       const OpensslCredentials_t * pOpensslCredentials );
+
+/*-----------------------------------------------------------*/
+
+static void logPath( const char * path,
+                     size_t pathLen,
+                     const char * fileType )
+{
+    char * cwd = getcwd( NULL, 0 );
```

Comment:
Should `getcwd` be called only in the else case?

Reply:
Good catch